docs/Guide/preconditions/creating-your-own-preconditions.mdx

```diff
@@ -1,6 +1,225 @@
+```
+
+</TabItem>
+<TabItem value="typescript" label="TypeScript">
+
+```typescript
+import { Precondition } from '@sapphire/framework';
+
+export class OwnerOnlyPrecondition extends Precondition {}
+```
+
+</TabItem>
+</Tabs>
+
+Next, we can create a [`run`][preconditionrun] function to execute our logic. This function should either return
+[`this.ok()`][preconditionok] to signify the condition has passed, or [`this.error(...)`][preconditionerror] to signify
+the command should be denied.
+
+<Tabs groupId="language-choice">
+<TabItem value="javascript" label="JavaScript" default>
+
+```javascript {4-8}
+const { Precondition } = require('@sapphire/framework');
+
+module.exports = class OwnerOnlyPrecondition extends Precondition {
+  public run(message) {
+    return message.author.id === 'YOUR_ID'
+      ? this.ok()
+      : this.error({ message: 'Only the bot owner can use this command!' })
+  }
+};
+```
+
+</TabItem>
+<TabItem value="ESM" label="ESM">
+
+```javascript {4-8}
+import { Precondition } from '@sapphire/framework';
+
+export class OwnerOnlyPrecondition extends Precondition {
+  public run(message) {
+    return message.author.id === 'YOUR_ID'
+      ? this.ok()
+      : this.error({ message: 'Only the bot owner can use this command!' })
+  }
+}
+```
+
+</TabItem>
+<TabItem value="typescript" label="TypeScript">
+
+```typescript {4-8}
+import { Precondition, Message } from '@sapphire/framework';
+
+export class OwnerOnlyPrecondition extends Precondition {
+  public run(message: Message) {
+    return message.author.id === 'YOUR_ID'
+      ? this.ok()
+      : this.error({ message: 'Only the bot owner can use this command!' });
+  }
+}
+```
+
+</TabItem>
+</Tabs>
+
+## Using Preconditions in Commands
+
+To attach a precondition to a command, you simply have to input its name in an array in the command's
+[`preconditions`][preconditions-option] option.
+
+<Tabs groupId="language-choice">
+<TabItem value="javascript" label="JavaScript" default>
+
+```javascript {7}
+const { Command } = require('@sapphire/framework');
+
+module.exports = class PingCommand extends Command {
+  constructor(context) {
+    super(context, {
+      // ...
+      preconditions: ['OwnerOnly']
+    });
+  }
+};
+```
+
+</TabItem>
+<TabItem value="ESM" label="ESM">
+
+```javascript {7}
+import { Command } from '@sapphire/framework';
+
+export class PingCommand extends Command {
+  constructor(context) {
+    super(context, {
+      // ...
+      preconditions: ['OwnerOnly']
+    });
+  }
+}
+```
+
+</TabItem>
+<TabItem value="typescript" label="TypeScript">
+
+```typescript {7}
+import { Command } from '@sapphire/framework';
+
+export class PingCommand extends Command {
+  public constructor(context: Command.Context) {
+    super(context, {
+      // ...
+      preconditions: ['OwnerOnly']
+    });
+  }
+}
+```
+
+</TabItem>
+</Tabs>
+
+Now, if someone who is not the bot owner executes the ping command, nothing will happen!
```

Comment:
```suggestion
Now, if someone who is not the bot owner executes the `ping` command, nothing will happen!
```

Reply:
Bumping this because I have no idea if it's actually resolved due to the merge markers